Data-Source-Templates.md

```diff
@@ -1,6 +1,107 @@
 # Data Source Templates
 
-This section will describe **Data Source Templates** in Cacti.
+Data sources serve an important role in Cacti data sources are data points 
+collected from the devices you monitor in Cacti.
+
+Data sources take the data that has been collected by the poller.
+
+The Data source then holds the value to be presented to the graph.
+
+The Data source template is then associated with a graph template.
+
+These data sources can be for example incoming traffic on an interface.
+Cacti allows you to create data source templates to associate to data queries
+the template allows you to describe what type of data Cacti should expect.
+
+To create a Data source you will need to select the proper data input there are several options 
+but the most popular ones would be GET SNMP data or GET SNMP data (indexed.)
+
+You would use Get SNMP data (indexed) for interfaces for example as Cacti would need to query the device to find the interface index.
+You would use Get SNMP data and put in a single OID for an OID that won't change for example CPU usage 
+generally does not have an index and the OID won't change
+
+You will also need to tell the template what the data source type is. 
+The most common ones being Gauge and Counter.
+These data types come from the RRDtool its self here are the following supported data types and their uses
+
+(Source https://oss.oetiker.ch/rrdtool/doc/rrdcreate.en.html)
+
+
+GAUGE
+
+    is for things like temperatures or number of people in a room or the value of a RedHat share.
+
+COUNTER
+
+    is for continuous incrementing counters like the ifInOctets counter in a router. 
+    The COUNTER data source assumes that the counter never decreases, except when a counter overflows.
+    The update function takes the overflow into account. The counter is stored as a per-second rate. 
+    When the counter overflows, RRDtool checks if the overflow happened at the 32bit or 64bit border 
+    and acts accordingly by adding an appropriate value to the result.
+
+DCOUNTER
+
+    the same as COUNTER, but for quantities expressed as double-precision floating point number. 
+    Could be used to track quantities that increment by non-integer numbers, i.e. number of seconds that some routine has taken to run,
+    total weight processed by some technology equipment etc. 
+    The only substantial difference is that DCOUNTER can either be upward counting or downward counting, but not both at the same time. 
+    The current direction is detected automatically on the second non-undefined counter update and any further change in the direction is considered a reset. 
+    The new direction is determined and locked in by the second update after reset and its difference to the value at reset.
+
+DERIVE
+
+    will store the derivative of the line going from the last to the current value of the data source. 
```

Comment:
Although this is continuing from the section title, it should repeat the word not just continue as a sentence.